### DISCUSSION TITLE
homebrew-core/Formula/sdl2_sound.rb not using provided internals

### DISCUSSION THREAD

**Discussion by ericoporto** [Homebrew/discussions#3788] [Everyday usage]
The main reason for using SDL2_sound vs anything else, is that it's self contained. But for some reason the package here is not using the internally provided and separately maintained timidity and libmodplug. So if you statically link to it hoping to get your app to someone, you will get an unexpected dependency and everything you did will break for no good reason.

Would it be possible to keep the original internals, so statically linking to it has the same usefulness as when doing so when building from source?

**Comment by Bo98** [comment]
What you described is pretty standard for a package manager, and can usually be fixed by including the static libraries of dependencies (typically pkg-config files list these under `Libs.private` so `pkg-config --static --libs` can be called to get the correct LDFLAGS). Bundled dependencies is something we always try to avoid, like many other package mangers also do - you will see the same situation on Ubuntu, Debian, Fedora etc.

However in the case of SDL2_Sound, I don't think there's an issue to begin with. Although the formula is listing those dependencies, it's not actually using them because upstream don't support external dependencies at the moment:

```
$ otool -L /usr/local/opt/sdl2_sound/lib/libSDL2_sound.dylib
/usr/local/opt/sdl2_sound/lib/libSDL2_sound.dylib:
	/usr/local/opt/sdl2_sound/lib/libSDL2_sound.2.dylib (compatibility version 2.0.0, current version 2.0.1)
	/usr/local/opt/sdl2/lib/libSDL2-2.0.0.dylib (compatibility version 23.0.0, current version 23.0.0)
	/System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox (compatibility version 1.0.0, current version 1000.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
```

So while declaring the dependencies was likely a mistake given they are unused, they shouldn't be affecting your use of static linking at all.

**Reply by ericoporto** [reply]
The sdl2_sound improvement over sdl_sound is not requiring any dependencies other than sdl2, forcing unecessary "repack" of dependencies defeats this and instead makes sure people using this library will need to place it's source files directly in their projects instead to ensure things workout as intended. I would really prefer if original intent was preserved.

**Reply by Bo98** [reply]
> forcing unecessary "repack" of dependencies defeats this and instead makes sure people using this library will need to place it's source files directly in their projects instead to ensure things workout as intended. I would really prefer if original intent was preserved.

This is not what's happening. Can you detail precisely what issue you are seeing when you try use `sdl2_sound`?
